huaweicloud/services/apig/resource_huaweicloud_apig_vpc_channel.go

```diff
@@ -340,91 +410,97 @@ func setApigVpcChannelParameters(d *schema.ResourceData, config *config.Config,
 		d.Set("timeout", resp.VpcHealthConfig.Timeout),
 		d.Set("interval", resp.VpcHealthConfig.TimeInterval),
 		d.Set("http_code", resp.VpcHealthConfig.HttpCodes),
-		d.Set("create_time", resp.CreateTime),
-		d.Set("status", status),
-		setApigVpcChannelMemberType(d, resp.MemberType),
-		setApigVpcChannelAlgorithm(d, resp.BalanceStrategy),
-		setApigVpcChannelMembers(d, resp.MemberType, resp.Members),
+		d.Set("created_at", resp.CreateTime),
 	)
-	if mErr.ErrorOrNil() != nil {
-		return mErr
+
+	status, ok := channelStatus[resp.Status]
+	if !ok {
+		return diag.Errorf("the response status is invalid")
+	} else {
+		mErr = multierror.Append(mErr, d.Set("status", status))
+	}
+	// Extract the member type.
+	memberType := extractVpcChannelMemberType(resp.MemberType)
+	if memberType != nil {
+		mErr = multierror.Append(mErr, d.Set("member_type", string(*memberType)))
+	}
+	// Extract the member type.
```

Comment:
// Extract the member type.
// Extract the algorithm type.

Reply:
fixed.